### PR TITLE
changes to the IoT pages

### PR DESCRIPTION
**Pull request by MaximRouiller**

### CODE REVIEW

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\msri-o-unknown\raw-\pkgidyok.kyh\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\msri-o-unknown\raw-\pkgidyok.kyh\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C21%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707211158083200-249%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\qbua-o-5cc49f16\raw-\1pjzzzhu.xpx\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\qbua-o-5cc49f16\raw-\1pjzzzhu.xpx\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C21%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707211221131972-249%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: api/overview/azure/devices.md [hunk 1]
```diff
@@ -1,83 +0,0 @@
----
```
Comment (JeremyLikness) [inline api/overview/azure/devices.md]: I would check with @Campsoper on this - not sure we can just delete a file, but you may have already looked into this

File: api/overview/azure/iot.md [hunk 2]
```diff
@@ -17,40 +17,75 @@ ms.service: multiple
...
+### Examples 
 
-### Example
-
---Example overview--
+This example connects to the IoT Hub and sends a message every 1 second.
 
 ```csharp
-/* Code goes here */
+int _messageId = 1;
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: You can probably remove these variable declarations and just use them explicitly in the samples to reduce the surface area

File: api/overview/azure/iot.md [hunk 3]
```diff
@@ -17,40 +17,75 @@ ms.service: multiple
...
+string deviceKey = "<deviceKey>";
+string deviceId = "<deviceId>";
+DeviceClient deviceClient = DeviceClient.Create("<IoTHubHostname>", new DeviceAuthenticationWithRegistrySymmetricKey(deviceId, deviceKey), TransportType.Mqtt);
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: I would make IoTHubHostname a variable and maybe break this into two lines.

File: api/overview/azure/iot.md [hunk 4]
```diff
@@ -17,40 +17,75 @@ ms.service: multiple
...
+
+while (true)
+{
+    var currentTemperature = 20 + Rand.NextDouble() * 15;
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: No var ... explicit types

File: api/overview/azure/iot.md [hunk 5]
```diff
@@ -17,40 +17,75 @@ ms.service: multiple
...
+> [Explore the management APIs](/dotnet/api/overview/azure/devices/client)
 
 ## Samples
 
---Samples list here--
+- [Generic Web Service to Event Hub scenario](https://azure.microsoft.com/resources/samples/event-hubs-dotnet-importfromweb/)
+
+View the [complete list](https://azure.microsoft.com/resources/samples/?platform=dotnet&service=iot-hub) of Azure IoT Hubsamples.
+
+View the [Azure IoT Hub developer guide](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide) for more guidances.
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: Probably should be "guidance" not plural ("guidances").

File: api/overview/azure/devices.md (hunk 1 above)
Comment (MaximRouiller) [inline api/overview/azure/devices.md]: already done and @CamSoper is already looped in.

File: api/overview/azure/iot.md [hunk 6]
```diff
@@ -17,40 +17,78 @@ ms.service: multiple
...
 ```bash
-dotnet add package Microsoft.Azure.Management.IotHub
+dotnet add package Microsoft.Azure.Devices.Client
 ```
+### Examples 
 
-### Example
-
---Example overview--
+This example connects to the IoT Hub and sends a message every 1 second.
```
Comment (johnpapa) [inline api/overview/azure/iot.md]: wording feels awkward. How about ...

> ... and sends a message once a second

or 

> ... and sends one message per second

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\ntrx-o-c534aef7\raw-\5zdgvm4t.khi\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\ntrx-o-c534aef7\raw-\5zdgvm4t.khi\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C21%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707211250379260-249%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: api/overview/azure/iot.md [hunk 7]
```diff
@@ -17,40 +17,78 @@ ms.service: multiple
...
+    double currentHumidity = 60 + Rand.NextDouble() * 20;
+
+    var telemetryDataPoint = new
+    {
+        messageId = _messageId++,
+        deviceId = deviceId,
+        temperature = currentTemperature,
+        humidity = currentHumidity
+    };
+    var messageString = JsonConvert.SerializeObject(telemetryDataPoint);
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: string

File: api/overview/azure/iot.md [hunk 8]
```diff
@@ -17,40 +17,78 @@ ms.service: multiple
...
+    var message = new Message(Encoding.ASCII.GetBytes(messageString));
```
Comment (JeremyLikness) [inline api/overview/azure/iot.md]: Message = new Message...

File: api/overview/azure/iot.md (hunk 6 above)
Comment (MaximRouiller) [inline api/overview/azure/iot.md]: agreed. I'm going with `one message per second`

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\wenh-o-528c9e65\raw-\edjmi2qv.yji\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\wenh-o-528c9e65\raw-\edjmi2qv.yji\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C21%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707211256492977-249%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by JeremyLikness** [comment]
#sign-off

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/MaximRouiller/azure-docs-sdk-dotnet/blob/aug-1-stage-iot/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\en00-o-da99ea10\raw-\xx4i4vkf.4dy\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\en00-o-da99ea10\raw-\xx4i4vkf.4dy\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C21%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707211309128408-249%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**